src/DOM/HTML/History.purs

```diff
@@ -0,0 +1,18 @@
+module DOM.HTML.History where
+
+import Control.Monad.Eff (Eff)
+import DOM.HTML.Types (HISTORY, History)
+import Data.Foreign (Foreign)
+import Prelude (Unit)
+
+-- DocumentTitle will set value of `document.title`
+newtype DocumentTitle = DocumentTitle String
+newtype Delta = Delta Int
+newtype URL = URL String -- Unsure how to better type this.
+
+foreign import back :: forall e. History -> Eff (history :: HISTORY | e) Unit
+foreign import forward :: forall e. History -> Eff (history :: HISTORY | e) Unit
+foreign import go :: forall e. History -> Delta -> Eff (history :: HISTORY | e) Unit
+foreign import pushState :: forall e. History -> Foreign -> DocumentTitle -> URL -> Eff (history :: HISTORY | e) Unit
+foreign import replaceState :: forall e. History -> Foreign -> DocumentTitle -> URL -> Eff (history :: HISTORY | e) Unit
+foreign import state :: forall e. History -> Eff (history :: HISTORY | e) Foreign
```

Comment:
Sorry I didn't notice this earlier, but can we put the `History` argument at the end for all of these? That's more consistent with the rest of the API, and is done so you can write things like:

``` purescript
go -2 =<< history =<< window
```